xls/dslx/bytecode/bytecode_interpreter_test.cc

```diff
@@ -1519,6 +1519,78 @@ fn main() {
   EXPECT_TRUE(value.IsUnit());
 }
 
+// GH 981
+// Assert_eq type deduction didn't fail in
+// the first iteration of the next function.
+// This test requires 2 iterations of next function,
+// which triggered assert_eq bug.
+TEST(BytecodeInterpreterTest, AssertEqFailProcs) {
```

Comment:
nit: what about naming `AssertEqFailProcIterations` to capture the iteration clause?